### PR TITLE
Fits writer master

### PR DESCRIPTION
**Pull request by MaxNoe**
I cherry picked @alexeyegorov commits to have the changes in the master, not the refactoring branch

I added a few minor changes, which make it possible to read in the written file with fact tools again.

This can now write fits files and read the result. One could write out photoncharges/arrivaltimes and do the analysis later.

### CODE REVIEW

**Comment by MaxNoe** [comment]
This FITS writer has some limitations:

* It keeps all the stuff in memory until the end of the stream (write operation happens in `finish`)
* It does not support variable length columns

**Comment by MaxNoe** [comment]
I added a new `Processor` called `StreamingFITSWriter` which writes out each data item directly and does not keep the whole table in RAM.

It sadly also does not support variable length arrays like for example `shower` pixels.

**Comment by MaxNoe** [comment]
What do you think: Should we keep both Processors or just the `StreamedFITSWriter`? Possibly as `FITSWriter`

**Comment by MaxNoe** [comment]
Also I do not like how all those type checks happen twice. Do have an idea how to do this better?

**Comment by MaxNoe** [comment]
Reading in the fits file using astropy ist ~50 times faster than using `json.load` on a json file containing the same data.

**Comment by MaxNoe** [comment]
@fact-project/fact-tools-core-dev 

I think this is ready for review now. Any comments?

File: src/main/java/fact/io/FITSWriter.java [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+ *
+ * The fits file is initialised using the given keys from first data item.
+ * All following items must have the same structure.
+ *
+ */
+public class FITSWriter implements StatefulProcessor {
+
+    static Logger log = LoggerFactory.getLogger(FITSWriter.class);
+
+    @Parameter(required = true)
```
Comment (jebuss) [inline src/main/java/fact/io/FITSWriter.java]: description?

File: src/main/java/fact/io/FITSWriter.java [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+    public void setKeys(Keys keys) {
+        this.keys = keys;
+    }
+
+    public void setUrl(URL url) {
+        this.url = url;
+    }
+
+    public void setExtname(String extname) {
```
Comment (jebuss) [inline src/main/java/fact/io/FITSWriter.java]: setExtName

**Comment by jebuss** [comment]
LGTM

**Pull request merged**